diff --git a/src/SDL12_compat.c b/src/SDL12_compat.c
index abc9e01..dccb57c 100644
--- a/src/SDL12_compat.c
+++ b/src/SDL12_compat.c
@@ -1341,7 +1341,12 @@ static char loaderror[256];
     }
 #elif defined(__unix__)
     #include <dlfcn.h>
+#ifdef __stanix__
+    // Stanix seem to have a different library name
+    #define SDL20_LIBNAME "libSDL2-2.0.so"
+#else
     #define SDL20_LIBNAME "libSDL2-2.0.so.0"
+#endif
     #define SDL20_REQUIRED_VER SDL_VERSIONNUM(2,0,7)
     static void *Loaded_SDL20 = NULL;
     #define LoadSDL20Library() ((Loaded_SDL20 = dlopen(SDL20_LIBNAME, RTLD_LOCAL|RTLD_NOW)) != NULL)
