diff --git a/makefile b/makefile
index 0669744..b5bddfc 100644
--- a/makefile
+++ b/makefile
@@ -8,9 +8,11 @@ CFLAGS += -DHOST="${HOST}" ${OPT}
 
 all :  ${EXE}
 % : src/%.c
-	${CC} ${CFLAGS} -o $@ $^
+	${CC} ${CFLAGS} -o $@.o -c $^
+	#we need to make a raw call to LD
+	${LD} ${LDFLAGS} -o $@ $@.o ${SYSROOT}/usr/lib/crt0.o -L ${SYSROOT}/usr/lib -lc
 clean :
-	rm -f ${EXE}
+	rm -f ${EXE} *.o
 install : ${EXE}
 	cp ${EXE} ${PREFIX}/bin
 config.mk :
diff --git a/src/ls.c b/src/ls.c
index 1750c23..7af9609 100644
--- a/src/ls.c
+++ b/src/ls.c
@@ -32,7 +32,7 @@ void list_add(char *str){
 	}
 	entry_count++;
 	list = realloc(list,entry_count * sizeof(char *));
-	list[entry_count-1] = str;
+	list[entry_count-1] = strdup(str);
 }
 
 void info(char *path){
diff --git a/src/test.c b/src/test.c
index a1d226d..8dddc9b 100644
--- a/src/test.c
+++ b/src/test.c
@@ -4,6 +4,12 @@
 #include <unistd.h>
 
 //simple test command
+uid_t geteuid(){
+	return 0;
+}
+gid_t getegid(){
+	return 0;
+}
 
 int main(int argc,char **argv){
 	if(argc < 3){
